plugins/components/controllable/Controllable.hpp

```diff
@@ -160,18 +171,24 @@ namespace Components {
          * @throws std::runtime_error If the data size is invalid.
          */
         void deserialize(std::vector<uint8_t> &data) override {
-            if (data.size() != sizeof(__data))
+            if (data.size() != getSize())
                 throw std::runtime_error("Invalid data size for Controllable component");
             std::copy(data.begin(), data.begin() + 4, inputs.begin());
             std::copy(data.begin() + 4, data.end(), actions.begin());
+
+            size_t index = sizeof(__data);
+            for (const auto &pair : keyBindings) {
+                Key key = static_cast<Key>(data[index++]);
+                keyBindings[pair.first] = key;
+            }
         };
 
         /**
          * @brief Gets the size of the serialized data.
          * 
          * @return The size of the data, in bytes.
          */
-        size_t getSize() const override { return sizeof(__data); };
+        size_t getSize() const override { return 28; };
```

Comment:
Why return a hardcoded value instead of the previous `sizeof` expression?

Reply:
Ive changed it to be dynamic now